packages/playwright-core/src/server/isomorphic/utilityScriptSerializers.ts

```diff
@@ -68,6 +71,42 @@ export function source() {
     }
   }
 
+  const typedArrayCtors: Record<TypedArrayKind, Function> = {
+    i8: Int8Array,
+    ui8: Uint8Array,
+    ui8c: Uint8ClampedArray,
+    i16: Int16Array,
+    ui16: Uint16Array,
+    i32: Int32Array,
+    ui32: Uint32Array,
+    // TODO: add Float16Array once it's in baseline
+    f32: Float32Array,
+    f64: Float64Array,
+    bi64: BigInt64Array,
+    bui64: BigUint64Array,
+  };
+
+  function typedArrayToBase64(array: any) {
+    if (globalThis.Buffer)
```

Comment:
Why have Node-specific code? It seems like `btoa/atob` should work both in the browser and in the Node.

Reply:
good question! I mostly copied from @pavelfeldman's gist. Pavel, why?

Reply:
looks like it works just fine without: https://github.com/microsoft/playwright/pull/34949/commits/dfd44ee743424a63e096e63214f58907c49e577d